src/thread_task.rs

```diff
@@ -1295,35 +1271,35 @@ pub unsafe fn rav1d_worker_task(c: &Rav1dContext, task_thread: Arc<Rav1dTaskCont
                                 ttd.cond.notify_one();
                             }
                         }
-                        task_type = RAV1D_TASK_TYPE_SUPER_RESOLUTION;
+                        task_type = TaskType::SuperResolution;
                         continue 'fallthrough;
                     }
-                    RAV1D_TASK_TYPE_SUPER_RESOLUTION => {
+                    TaskType::SuperResolution => {
                         let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
                         if frame_hdr.size.width[0] != frame_hdr.size.width[1] {
                             if f.task_thread.error.load(Ordering::SeqCst) == 0 {
                                 (f.bd_fn().filter_sbrow_resize)(c, f, &mut tc, sby);
                             }
                         }
-                        task_type = RAV1D_TASK_TYPE_LOOP_RESTORATION;
+                        task_type = TaskType::LoopRestoration;
                         continue 'fallthrough;
                     }
-                    RAV1D_TASK_TYPE_LOOP_RESTORATION => {
+                    TaskType::LoopRestoration => {
                         if f.task_thread.error.load(Ordering::SeqCst) == 0
                             && f.lf.restore_planes != 0
                         {
                             (f.bd_fn().filter_sbrow_lr)(c, f, &mut tc, sby);
                         }
-                        task_type = RAV1D_TASK_TYPE_RECONSTRUCTION_PROGRESS;
+                        task_type = TaskType::ReconstructionProgress;
                         continue 'fallthrough;
                     }
-                    RAV1D_TASK_TYPE_RECONSTRUCTION_PROGRESS => {
+                    TaskType::ReconstructionProgress => {
                         // dummy to cover for no post-filters
                     }
-                    RAV1D_TASK_TYPE_ENTROPY_PROGRESS => {
+                    TaskType::EntropyProgress => {
                         // dummy to convert tile progress to frame
                     }
-                    _ => {
+                    TaskType::FgPrep | TaskType::FgApply => {
```

Comment:
I think this should remain `_`, as it was a `default:` in C.

Reply:
I assumed that was because there are no exhaustive matches in C. I don't see any downsides to the exhaustive match here 

- when you read the code, it's instantly clear what cases are left over and handled by this fallback
- when a new case gets added the exhaustiveness checker will ensure the match is updated
- when a case is removed it's clear what to do

there are cases where making a match exhaustive is very verbose, but that's not the case here either

Reply:
Hmm, if you've thought about it carefully, I think it's okay to change then.  There's not really any risk of something going wrong here.